src/main/java/uk/gov/hmcts/ccd/data/casedetails/DefaultCaseDetailsRepository.java

```diff
@@ -207,13 +207,13 @@ private Query getQueryByParameters(MetaData metadata, final Map<String, String>
     }
 
     private Query getCountQueryByMetaData(MetaData metadata) {
-        return queryBuilderFactory.count(em)
+        return queryBuilderFactory.count(em, metadata)
                                   .whereMetadata(metadata)
```

Comment:
now that the factory gets a metadata, we could have the factory itself calling whereMetadata(metadata) on the builder, rather than doing it explicitly here

Reply:
Actually the factory should just return the builder and not do any building within. Do I have leverage to refactor this?

Reply:
reorganised code now